yt_dlp/extractor/soundcloud.py

```diff
@@ -656,6 +665,11 @@ def _extract_set(self, playlist, token=None):
                     'playlistId': playlist_id,
                     'playlistSecretToken': token,
                 }, headers=self._HEADERS)
+        album_info = traverse_obj(playlist, {
+            'album': ('title', {str}),
+            'album_artist': ('user', 'username', {str}),
+            'album_type': ('set_type', {str}),
```

Comment:
maybe we also wanna do this...
```suggestion
            'album_type': ('set_type', {str}, filter),
```

Reply:
IMO this is fine because this is the 'album type' according to SoundCloud - they use the empty string for 'playlist' type sets; it's not really the absence of an album type, just their way of telling playlists apart from albums.

Reply:
I guess it mainly depends on how other extractors use the 'album_type' field, I'm not too sure if there's a standard

Reply:
There's only one other extractor that currently returns the field afaik. I think keeping an empty string is useless and possibly confusing to anyone who doesn't know the workings of the soundcloud API

I'd either filter, or do something like this:

```suggestion
            'album_type': ('set_type', {str}, lambda x: x or 'album'),
```
```suggestion
            'album_type': ('set_type', {str}, lambda x: x or 'lp'),
```
etc

Reply:
I made it explicitly return 'playlist' for album type if its empty